### ISSUE TITLE
All apps using googleads-mobile-flutter getting blocked for "Ad Fraud"

### ISSUE THREAD

**Issue by dvird** [googleads/googleads-mobile-flutter#155]
I have 12 apps,

6 of flutter and 6 native.

all 6 of flutter getting blocked for ad fraud by AdMob, and all other native everything totally fine.
I'm doing the same in all apps, one interstitial ad on splash and that's it.

Maybe somehow it's the library? I can't find any reasonable solution for this issue

**Comment by hosamcr7** [comment]
I have the same issue, any solutions?

**Comment by bparrishMines** [comment]
cc @zoeyfan 

@dvird Which platform are you seeing this on? Can you also run `flutter doctor` so we can see which version of Flutter you are using?

**Comment by dvird** [comment]
No logs. no platform. everything looks fine it's very difficult to debug this.. 
doctor all pass. flutter 2.0.4 but also happen before. on previous admob-sdk and previous flutter

**Comment by zoeyfan** [comment]
@dvird Thanks for raising to our attention. Would you please let me know if this just happened recently, or started as soon as you implemented the plugin?

**Comment by dvird** [comment]
Hi thanks for you’re attention.
As I said it happened also in last sdk - firebase_admob

Unfortunately I did not find any rules for this issue. All of the apps new ones, but some it’s blocked after 1 day and others after 1 month.

**Comment by zoeyfan** [comment]
If I understand correctly, you recently published 6 new Flutter apps (roughly 1 month ago). You used googleads-mobile-flutter, but all your 6 apps got blocked due to "Ad fraud"? You switched to "firebase_admob", and you are still blocked for the same reason? I think we will need some specific information about your app, app id, and publisher id to further look into this. Do you feel comfortable providing those to us?

**Comment by whiplashoo** [comment]
Seeing a similar issue on one of my apps. Currently, I have one banner ad and one interstitial on user action.

Received **ad serving penalty** from Google AdMob for **"invalid traffic"**. Now **no ads seem to load** and when I run in Simulator, I get: 

`InterstitialAd failed to load.  Too many recently failed requests for ad unit ID: ca-app-pub-(my-publisher-id). You must wait a few seconds before making another ad request. `

Maybe it has any relation to [this recent change](https://groups.google.com/g/google-admob-ads-sdk/c/_f0YqR1lYt0)? 

flutter doctor -v
```
[✓] Flutter (Channel stable, 2.0.4, on Microsoft Windows
    [Version 10.0.19042.867], locale el-GR)
    • Flutter version 2.0.4 at C:\src\flutter
    • Framework revision b1395592de (10 days ago),
      2021-04-01 14:25:01 -0700
    • Engine revision 2dce47073a
    • Dart version 2.12.2

[✓] Android toolchain - develop for Android devices
    (Android SDK version 30.0.2)
    • Android SDK at
      C:\Users\whipl\AppData\Local\Android\sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: C:/Program Files/Android/Android
      Studio\jre\bin\java
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b01)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at C:\Program
      Files\Google\Chrome\Application\chrome.exe

[✓] Android Studio (version 4.1.0)
    • Android Studio at C:\Program Files\Android\Android
      Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b01)

[✓] Android Studio
    • Android Studio at C:/Program Files/Android/Android
      Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = C:/Program
      Files/Android/Android Studio
    • Java version OpenJDK Runtime Environment (build
      1.8.0_242-release-1644-b01)

[✓] Connected device (2 available)
    • Chrome (web) • chrome • web-javascript • Google
      Chrome 89.0.4389.114
    • Edge (web)   • edge   • web-javascript • Microsoft
      Edge 88.0.705.63

• No issues found!
```

Also my app is fully migrated to null-safety.

**Comment by vishalknishad** [comment]
For me both of these (flutter_native_admob, firebase_admob) cause limited ad serving issue and now I'm seeing this.

**Comment by dvird** [comment]
@zoeyfan give me private communication ill share that with you. I'm telling you it's a catastrophic issue. I'm losing a lot of money because of that and no one to speak with.

on google ads where I pay money that have the full support and they answer within 1 min..

**Comment by zoeyfan** [comment]
> @zoeyfan give me private communication channel ill share that with you. I'm telling you it's a catastrophic issue. I'm losing a lot of money because of that and no one to speak with.
> 
> on google ads where I pay money they have the full support and they answer within 1 min..



> @zoeyfan give me private communication channel ill share that with you. I'm telling you it's a catastrophic issue. I'm losing a lot of money because of that and no one to speak with.
> 
> on google ads where I pay money they have the full support and they answer within 1 min..

Please provide your email address and I will follow up from there.

**Comment by zoeyfan** [comment]
@dvird would you please provide your email so I can follow up with you to understand what's going on with your account?

**Comment by dvird** [comment]
As I told you I dont to send it here. give me a private channel to send it to you.. discord or email or something

**Comment by zoeyfan** [comment]
@dvird can you DM me here? https://twitter.com/ZoeyFan723

**Comment by dvird** [comment]
@zoeyfan dm is blocked on your account

**Comment by flutterDev-sudo** [comment]
My team has been working on a flutter app for the last 7 months and we are shocked and feel terrible to see that this is an issue 😢 , we spent so much money & time on our app and our main primary source of income is suppose to be admob. We will not publish our app although it is tested and ready for release very soon as long as this issue is not fixed, please keep this thread updated and solve this issue asap. We would kindly appreciate it!

**Comment by dvird** [comment]
@flutterDev-sudo joined 4 min ago? sounds promising

**Comment by flutterDev-sudo** [comment]
@dvird I do not want to provide my original github account as I do not want to leak personal information. I have no reason to lie or mislead. I have been checking this thread ever since it is posted. I decided to create a burner account because our app is nearly fully ready for release but this issue is blocking us.

**Comment by zoeyfan** [comment]
@dvird unblocked. Please try again.

**Comment by dvird** [comment]
sent you. please take a look. thanks

**Comment by zoeyfan** [comment]
@flutterDev-sudo I understand your hesitation about publishing your app given the discussion in this thread. To be honest, given the feedback we have been receiving from other customers, the plugin is indeed working as expected. I am in the process of figuring out what settings in dvird's apps are trigger this issue. But based on the current data we have, we cannot conclude that this is an issue with the plugin. 

Nevertheless, we will keep this thread updated as we have more information.

**Comment by vishalknishad** [comment]
@zoeyfan Does calling setState multiple times result in such issues? 
I don't think so because ads are initialized in initState. What's your thought on this?

**Comment by zoeyfan** [comment]
@bparrishMines do you know the question above?

**Comment by bparrishMines** [comment]
@whiplashoo If you're attempting to get the Ads to show in a simulator, I'm assuming you're using the `adUnitId` from your adMob account. If so, this should only be done in the release version of your app and test `adUnitId` should be used for testing and debugging. 

@vishalknishad `setState` shouldn't cause a new `Ad` to be loaded. If it does, you should see the `AdListener.onAdLoaded()` callback called multiple times.

@dvird @flutterDev-sudo @vishalknishad Is this error seen on Android, iOS, or both? Also, attempting to test your app with non test `adUnitId`s could potentially cause this problem. See https://github.com/googleads/googleads-mobile-flutter#always-test-with-test-ads

**Comment by bparrishMines** [comment]
I would also like everyone to remind everyone of [Flutter's Code of Conduct](https://github.com/flutter/flutter/blob/master/CODE_OF_CONDUCT.md) as we continue to debug this issue.

**Comment by dvird** [comment]
Currently, I've checked that only on android apps, so I can't really tell how is it working there.
@bparrishMines thanks for your support

**Comment by whiplashoo** [comment]
@bparrishMines Yes, you are right, should have tested with the test ids. Disregard that.

However, the ad serving limit in production is still active, so I get no ads shown at all. I will wait to see if it will be eventually lifted, as I am not aware of any policy violation on my end.

**Comment by vishalknishad** [comment]
@bparrishMines I tested with test ids and succesfully made to production with real ad ids. The ads were displaying fine for 2 weeks and then the issue suddenly arose.

**Comment by mcradha** [comment]
I am facing a similar issue after I migrated to google_mobile_ads plugin from firebase_admob (which was [deprecated](https://pub.dev/documentation/firebase_admob/latest/) earlier in March) . Earlier with Firebase admob I was able to show Rewarded ads. Now I keep getting an error which says 

"[ERROR:cookie_manager.cc(137)] Strict Secure Cookie policy does not allow setting a secure cookie for http://googleads.g.doubleclick.net/ for apps targeting >= R. Please either use the 'https:' scheme for this URL or omit the 'Secure' directive in the cookie value."

I have also got ad limiting placed on my admob account after I integrated google_mobile_ads plugin for Flutter.

**Comment by zoeyfan** [comment]
Thank you for your patience while we are looking into this. The googleads-mobile-flutter plugin is a wrapper around the GMA SDK. We have done a thorough investigation internally with several teams and confirmed the reported issue was not caused by the googleads-mobile-flutter plugin.

Google is constantly investing in measures to protect and improve the experience for advertisers, publishers, and users of AdMob.

As we have been unable to verify the quality of your traffic, your ad serving is currently being limited. Your sites or apps may experience less demand from Google while we continue to evaluate your account compliance and traffic quality.

We understand that you may want to know more about the activity we’ve detected, but in order to ensure we maintain a healthy ecosystem for all stakeholders we are unable to provide you with information about specific account activity. While this ad serving limit typically impacts publishers for less than 30 days, it may take longer in some cases. Unfortunately, we can’t say how long this will take in your case.

You can view details about this account level enforcement in your Policy center, and learn more about ad serving limits and what you can do to address them here. 

In the meantime, your account is still fully accessible and you can view details about this account-level issue in the Policy center. We encourage you to be proactive in making sure your account complies with our AdMob and AdSense Program policies, as well as checking out our Help Center on Invalid Traffic to learn about how to prevent invalid traffic and improve your traffic quality.

Thank you for your understanding and cooperation.

**Comment by marvinJ10** [comment]
Hi everyone.
My app just got rejected after an update using the latest version - google_mobile_ads 0.13.0 
Below is the console reply 

Issue with your app
We have determined your app contains code to facilitate Ad Fraud.
Affected Versions:
App Bundle:16

My implementation is same as shown in google_mobile_ads 0.13.0 package example. 

Any leads on how to fix this issue?

**Comment by mk-dev-1** [comment]
Morning everyone,

apologies on jumping in here with my own issue, which may or may not be related. Our app did pass app review and did not get blocked, however our AdMob account has been put under investigation for invalid clicks. In our app we show a preloaded interstitial after the user has clicked the save button, I.e. finished the given task and therefor has reached a break point. When calling `show()` on the interstitial, we are *not* waiting for the interstitial to close, but execute `Navigator.of(context).pop()` immediately, so that that the navigation happens in the background.

Reading all of the above and other threads, I am concerned that this may be the cause of algorithms flagging our app as it may constitute a policy violation. Our 30 day investigation period is almost up and we have not made any changes so far, so getting banned is reasonably likely at the moment. Traffic, clickrates and all of the likes are not showing any warning signs...

Can anyone share some thoughts on this? We are a little bit desperate as Google is not responding to any inquiries on this or sharing any helpful information.

Thank you so much!

**Comment by dvird** [comment]
@zoeyfan @bparrishMines guys, please. there is an issue with the library, im keep uploading apps and they are keep getting blocked. please do not ignore it. go check it with your BI guys and check how many apps using flutter admob library getting blocked.

**Comment by vishalknishad** [comment]
Hi guys, 

I guess there is no problem with the package. It might be an issue from admob itself. Check on internet, thousands of people are suffering due to this.
I would suggest to not depend on a single ad network and go for mediation instead.

**Comment by liondev7** [comment]
hi,

TLDR : remove native ad units, there is something wrong with them in the implementation.

Details:

i had exactly the same problem, after doing many tests i think i found the issue for my case, it's the native ads, there is definitely something wrong with the native ads implementation with this package, that makes google think you're doing something wrong or trying to do ad fraud even if you do all best practices.

Here is what worked for me, remove all ad units from Admob and from your app, wait for a week or so for the ad serving limit to be lifted, then create your ad units back and put back ads into your app BUT DO NOT USE NATIVE ADS, i figured this out the hard way, when i put back the ads (interstitial + native) i get the ad serving limited, when i removed native and ONLY put back interstitials, it's working fine for me so far.

(NB : i'm implementing all best practices as well, testID...etc and traffic is organic).

**Comment by hortigado** [comment]
> Hola a todos. Mi aplicación acaba de ser rechazada después de una actualización con la última versión - google_mobile_ads 0.13.0 A continuación se muestra la respuesta de la consola
> 
> Problema con tu aplicación Hemos determinado que tu aplicación contiene código para facilitar el fraude publicitario. Versiones afectadas: App Bundle:16
> 
> Mi implementación es la misma que se muestra en google_mobile_ads ejemplo de paquete 0.13.0.
> 
> ¿Alguna pista sobre cómo solucionar este problema?

Same case :(

**Comment by liondev7** [comment]
There is definitely something wrong with this package, whenever used admob account gets limited with "invalid traffic concers" while traffic is 100% organic and/or bought via google ads, also implementation is 100% according to the examples in the documentation.

**Comment by bparrishMines** [comment]
As our team isn't taking any more action on this specific issue, I'm going to go ahead and close it. See https://github.com/googleads/googleads-mobile-flutter/issues/155#issuecomment-820617086 for more information about traffic and policies or follow steps to open a new issue.
